### ISSUE TITLE
Feeding Upgrade Dupe Bug

### ISSUE THREAD

**Issue by Blzd87** [Salandora/SophisticatedBackpacks#93]
Describe the bug
When the Feeding Upgrade in the Sophisticated Backpacks mod auto-feeds the player, if the player’s hotbar is highlighting a tool or weapon, the mod does not auto-feed. Instead, it duplicates the item currently in the player’s hand.

To Reproduce

Equip a backpack with the Feeding Upgrade installed.
Ensure there is food in the backpack.
Hold a tool or weapon in the hotbar and highlight it.
Let the Feeding Upgrade try to auto-feed the player (e.g., take damage or sprint until hunger drops).
Observe the duplication of the item instead of the auto-feeding.
Expected behavior
The Feeding Upgrade should auto-feed the player regardless of what is highlighted in the hotbar, without duplicating the item in hand.

Screenshots
If possible, add screenshots or a video showing the item duplication.

Versions

Mod version: 1.20.1-3.20.17.1.60
Forge version: (insert version here)
Minecraft version: 1.20.1
Additional context
No other mods seem to conflict with this feature, or list mods if applicable:

(list of installed mods)

**Comment by Salandora** [comment]
I'll investigate the problem.

Can you tell me what sophisticated core version you have as the problematic code is in there.

**Comment by Salandora** [comment]
So I tested this with Sophisticated Core 1.20.1-0.7.12.2.96 and Sophisticated Backpacks 1.20.1-3.20.17.1.60 and can not reproduce the issue. 
Is it reliably reproduceable for you? 
What other mods do you have to maybe add or change how food works?

**Comment by Blzd87** [comment]
Ah ok well i got two versions behind this one... I'll update asap and then see if the issue continues. but going by your testing it should be fixed in the newer versions.

**Comment by Salandora** [comment]
Where you able to reproduce the issue in the newer versions?

**Comment by Blzd87** [comment]
It still dupes. It dupes weapons. Swords and or bows. If your actively holding them at the time the feeding upgrade activates then it starts duping the weapon item into your backpack and when it runs out of empty slots it replaces your current slots which means it removes anything else you have any you start losing other items.

**Comment by Salandora** [comment]
Is this reproducible in instance with only Sophisticated Core and Backpacks? 
I have a feeling this is some incompatibility with another mod.

**Comment by Blzd87** [comment]
Sorry. I've got Too Many Bows mod. Which it dupes the most when holding a bow from that mod but it has also duped a diamond sword also

**Comment by Salandora** [comment]
Yeah but can you reproduce the issue with only Sophisticated mods? No other mods installed.

**Comment by Blzd87** [comment]
I will test this asap. and let you know what I find. As Im assuming theres no way to add support for custom items?

**Comment by Blzd87** [comment]
I've tested it with JUST the backpack / storage / core mod with vanilla sword and bow.

When holding the bow it does show the RIGHT CLICK action for drawing the arrow for a split second then stops and then your player is fed. so maybe this is causing a glitch/bug with the custom bows from Too Many Bows mod...

**Comment by Salandora** [comment]
Hmm interesting, I'll try to reproduce it with a bow and see if I can get the animation to play

**Comment by iHestia** [comment]
Currently playing a modpack and I encountered a similar issue, when you have a backpack with a feeding upgrade equipped and it tries to feed you and you're holding a chestplate you'll start swapping between the chestplate and the backpack, while also somehow creating duplicates. I assume Blzd87 had some sort of backslot mod allowing you to put weapons and bows in an armor slot. It also works with other armor/equippable slots, try 2 different pairs of boots, it swaps between them and puts the pairs in the backpack you're wearing. I tried with the off-hand as well by switching my off-hand button to right click, but that yielded no results. 
If a backpack duplicates it will be completely empty.

**Comment by Salandora** [comment]
Okay I just uploaded new versions, could you check if this keeps happening? 
Unfortunately I was never able to reproduce it myself so I need some feedback here @Blzd87 @iHestia

**Comment by iHestia** [comment]
Just tested it on 1.21.1 with the new version of the mod(3.23.4.1.96 and 1.2.9.1.140)  and it seems like the issue is fixed! On the older version of the mod it seems like the mod right clicks when the auto feeding upgrade does its thing, doesn't seem to happen any more though.

**Comment by Salandora** [comment]
Awesome thanks for letting me know.
I'll close the report then.
If for some reason it shows up again don't hesitate to make a new report.
